Subject: [PATCH] powerpc: Propagate regs through to machine_crash_shutdown
From: Michael Ellerman <michael@ellerman.id.au>
Date: 1133681952 +1100

Currently machine_crash_shutdown() gets a struct pt_regs, but doesn't pass it
through to the ppc_md function, it should.

Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
Signed-off-by: Paul Mackerras <paulus@samba.org>
Signed-off-by: Olaf Hering <olh@suse.de>

---

 arch/powerpc/kernel/machine_kexec.c |    2 +-
 include/asm-powerpc/machdep.h       |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

d718ab271e23c0629dd4caa02d0baa3e8ea7d16b
diff --git a/arch/powerpc/kernel/machine_kexec.c b/arch/powerpc/kernel/machine_kexec.c
index d8225c7..a91e40c 100644
--- a/arch/powerpc/kernel/machine_kexec.c
+++ b/arch/powerpc/kernel/machine_kexec.c
@@ -23,7 +23,7 @@ note_buf_t crash_notes[NR_CPUS];
 void machine_crash_shutdown(struct pt_regs *regs)
 {
 	if (ppc_md.machine_crash_shutdown)
-		ppc_md.machine_crash_shutdown();
+		ppc_md.machine_crash_shutdown(regs);
 }
 
 /*
diff --git a/include/asm-powerpc/machdep.h b/include/asm-powerpc/machdep.h
index d6a1a2b..3253902 100644
--- a/include/asm-powerpc/machdep.h
+++ b/include/asm-powerpc/machdep.h
@@ -222,7 +222,7 @@ struct machdep_calls {
 	 * to run successfully.
 	 * XXX Should we move this one out of kexec scope?
 	 */
-	void (*machine_crash_shutdown)(void);
+	void (*machine_crash_shutdown)(struct pt_regs *regs);
 
 	/* Called to do what every setup is needed on image and the
 	 * reboot code buffer. Returns 0 on success.
-- 
1.0.6
